hydrolib/core/basemodel.py

```diff
@@ -95,31 +96,57 @@ class FileModel(BaseModel, ABC):
     it doesn't error, but actually initializes the `FileModel`.
     """
 
+    __slots__ = ["__weakref__"]
+    # Use WeakValueDictionary to keep track of file paths with their respective parsed file models.
+    _file_models_cache: WeakValueDictionary = WeakValueDictionary()
     filepath: Optional[Path] = None
 
+    def __new__(cls, filepath: Optional[Path] = None, *args, **kwargs):
+        """Creates a new model.
+        If the file at the provided file path was already parsed, this instance is returned.
+
+        Args:
+            filepath (Optional[Path], optional): The absolute file path to the file. Defaults to None.
+
+        Returns:
+            FileModel: A file model.
+        """
+
+        if filepath:
+            filepath = Path(filepath)
+
+            if filepath in FileModel._file_models_cache:
+                logger.info(
+                    f"Returning existing FileModel from cache, because {filepath} was already parsed"
```

Comment:
You could use class name here instead of FileModel, so users see the actual (sub)class used. You could also add a . to the end of the sentence.

Reply:
Done